crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1011,25 +1059,12 @@ impl<'db> TypeInferenceBuilder<'db> {
     }
 
     fn module_ty_from_name(
-        &mut self,
-        module_name: Option<ModuleName>,
-        node: AnyNodeRef,
-    ) -> Type<'db> {
-        let Some(module_name) = module_name else {
-            return Type::Unknown;
-        };
-
-        if let Some(module) = resolve_module(self.db, module_name.clone()) {
-            Type::Module(module.file())
-        } else {
-            self.add_diagnostic(
-                node,
-                "unresolved-import",
-                format_args!("Import '{module_name}' could not be resolved."),
-            );
-
-            Type::Unknown
-        }
+        &self,
+        module_name: ModuleName,
+    ) -> Result<Type<'db>, ModuleResolutionError> {
```

Comment:
Again here I don't really see the value of returning a `Result` vs just emitting the diagnostic directly in this method.